src/routes/posts.js

```diff
@@ -1,38 +1,36 @@
 const express = require('express')
-const { logger } = require('../lib')
 const isAuthenticated = require('../middleware/isAuthenticated')
-
+const { catchAsync} = require('../lib')
 const postService = require('../services/posts')
+const {
+	transformArticleResponse ,
+	transformGifResponse
+} = require('./common/transformers')
+
+const typeTransformMap = {
+	article : transformArticleResponse,
+	gif : transformGifResponse
+}
 
-const log = logger()
 const router = express.Router()
 
 
 // GET REQUESTS
 
 // GET ALL ARTICLES
 
-const fetchPosts = async (req, res, next) => {
-	try {
-		const feed = await postService.fetchPosts()
-
-		res.status(200).json({
-			status: 'success',
-			data: feed.map((article) => ({
-				id: article.id,
-				userId: article.userId,
-				title: article.title,
-				content: article.content,
-				image: article.image,
-				published: article.published,
-				createdOn: article.createdAt
-			}))
-		})
-	} catch (err) {
-		log.error(err.message)
-		next(err)
-	}
-}
+const fetchPosts = catchAsync( async(req, res) => {
+
+	const feed = await postService.fetchPosts()
+
+	res.status(200).json({
+		status: 'success',
+		data: feed.map((post) => ({
+			...typeTransformMap[post.type](post)
+		}))
```

Comment:
great work. no need to spread in an object though if there's no other property

```suggestion
		data: feed.map((post) => typeTransformMap[post.type](post))
```

Reply:
it was being underlined when i wasn't spreading it

Reply:
okay, for what reason was it being underlined?

Reply:
`Parsing error : Unexpected token [eslint]`

Reply:
yea its not really about spreading it or not, kindly check what might be the cause, you could share it here if its proving a blocker as well

Reply:
yes please

Reply:
done